pontoon/sync/formats/json_extensions.py

```diff
@@ -0,0 +1,187 @@
+"""
+Parser for the .json translation format as used by the WebExtensions API:
+https://developer.mozilla.org/en-US/Add-ons/WebExtensions/Internationalization
+
+See also:
+https://www.chromium.org/developers/design-documents/extensions/how-the-extension-system-works/i18n
+"""
+import codecs
+import copy
+import json
+import logging
+import os
+
+from collections import OrderedDict
+from jsonschema import validate
+from jsonschema.exceptions import ValidationError
+
+from pontoon.sync import SyncError
+from pontoon.sync.exceptions import ParseError
+from pontoon.sync.formats.base import ParsedResource
+from pontoon.sync.vcs.models import VCSTranslation
+
+
+log = logging.getLogger(__name__)
+
+SCHEMA = {
+    "type": "object",
+    "patternProperties": {
+        ".*": {
+            "type": "object",
+            "properties": {
+                "message": {
+                    "type": "string"
+                },
+                "description": {
+                    "type": "string"
+                },
+                "placeholders": {
+                    "type": "object",
+                    "patternProperties": {
+                        ".*": {
+                            "type": "object",
+                            "properties": {
+                                "content": {
+                                    "type": "string"
+                                },
+                                "example": {
+                                    "type": "string"
+                                }
+                            },
+                            "required": ["content"]
+                        }
+                    }
+                }
+            },
+            "required": ["message", "description"]
```

Comment:
this needs to be at the same level as the properties its constraining

Reply:
The nesting level seems to be fine?
http://json-schema.org/examples.html

It also breaks if I move it one level up.

Reply:
my bad - you're right